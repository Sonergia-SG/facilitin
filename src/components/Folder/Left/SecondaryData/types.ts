import {
  FolderMOAString,
  FolderMOEString,
  FolderSiteString,
} from '../../../../store/reducer/entities/types';

export interface FormFieldList {
  label: string;
  value: string | null;
  values: Array<{
    key: string;
    label: string;
  }>;
  key: keyof FolderMOAString | keyof FolderMOEString | keyof FolderSiteString;
  type: 'list';
}

export interface FormFieldDate {
  label: string;
  value: string | null;
  key: keyof FolderMOAString | keyof FolderMOEString | keyof FolderSiteString;
  type: 'date';
}

export interface FormFieldTextBicRule {
  format: 'bic';
}

export interface FormFieldTextIbanRule {
  format: 'iban';
}

export interface FormFieldTextPhoneRule {
  format: 'phone';
}

export type FormFieldTextRules =
  | FormFieldTextPhoneRule
  | FormFieldTextBicRule
  | FormFieldTextIbanRule;

export interface FormFieldText {
  label: string;
  value: string | null;
  key: keyof FolderMOAString | keyof FolderMOEString | keyof FolderSiteString;
  type: 'text';
  rules?: FormFieldTextRules;
}

export interface FormFieldNumber {
  label: string;
  value: string | null;
  key: keyof FolderMOAString | keyof FolderMOEString | keyof FolderSiteString;
  type: 'number';
}

export type FormFieldDef =
  | FormFieldDate
  | FormFieldText
  | FormFieldNumber
  | FormFieldList;

export interface FormSectionDef {
  label: string;
  type: 'section';
  fields: Array<FormFieldDef>;
}

export type FormDef = FormFieldDef | FormSectionDef;

export type GenericForms = Array<FormDef>;

export interface Forms {
  moa: GenericForms;
  moe: GenericForms;
  site: GenericForms;
}
