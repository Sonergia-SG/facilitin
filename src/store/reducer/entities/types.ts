import {
  LOGOUT,
  FOLDER_LOADED,
  LIST_LOADED,
  FOLDER_UPDATE_MOA_LOADED,
  FOLDER_UPDATE_CHECK_POINT_LOADING,
  FOLDER_UPDATE_CHECK_POINT_LOADED,
  FOLDER_UPDATE_CHECK_POINT_ERROR,
  COMMENTS_LIST_LOADED,
  FOLDER_FILE_LITIGE_LOADED,
} from '../../types';

export interface User {
  id_user: number;
  nom: string;
  prenom: string;
}

export interface Users {
  [index: number]: User;
}

export interface CommentSimple {
  id_log: number;
  id_user: number;
  date_log: string;
  message: string;
}

export interface Comment extends CommentSimple {
  user: number;
}

export interface CommentFull extends CommentSimple {
  user: User;
}

export interface Comments {
  [index: number]: Comment;
}

export type BooleanNumber = 0 | 1;

export interface SimpleFile {
  id_dp_file: number;
  id_file: number;
  id_dp_operation: number;
  id_dossierprime: number;
  is_devis: BooleanNumber;
  is_aat: BooleanNumber;
  is_facture: BooleanNumber;
  is_lettre_engagement: BooleanNumber;
  is_justif_preca: BooleanNumber;
  is_autres: BooleanNumber;
  is_aft: BooleanNumber;
  is_contrat_distrib: BooleanNumber;
  is_conv_tepcv: BooleanNumber;
  is_conv_regroupement: BooleanNumber;
  is_liste_depenses: BooleanNumber;
  is_attestation_tepcv: BooleanNumber;
  is_attestation_collectivite: BooleanNumber;
  is_etat_recapitulatif: BooleanNumber;
  is_ah: BooleanNumber;
  is_horodatage: BooleanNumber;
  is_subrogation: BooleanNumber;
  statut: number;
  litige?: BooleanNumber;
}

export interface File extends SimpleFile {}

export interface FileFull extends SimpleFile {}

export interface Files {
  [index: number]: File;
}

export interface MOA {
  moa_prenom: string;
  moa_nom: string;
  moa_fonction: string;
  moa_rue: string;
  moa_rue2: string;
  moa_cp: string;
  moa_ville: string;
  moa_denomination: string;
}

export interface FolderMOAString {
  moa_civilite: string;
  moa_nom: string;
  moa_prenom: string;
  moa_rue: string;
  moa_rue2: string;
  moa_cp: string;
  moa_ville: string;
  moa_pays: string;
  moa_tel: string;
  moa_tel_2: string;
  moa_mobile: string;
  moa_email: string;
  moa_bic: string;
  moa_iban: string;
  moa_siret: string;
  moa_denomination: string;
  moa_fax: string;
  moa_commentaire: string;
  moa_individu_email: string;
  moa_contact: string;
  moa_contact_mobile: string;
  moa_raison_sociale_siege: string;
  moa_siren_siege: string;
  moa_adresse_siege: string;
  moa_cp_siege: string;
  moa_ville_siege: string;
  moa_fonction: string | null;
  moa_individu_fonction: string | null;
  moa_beneficiaire_role: string | null;
}

export interface SimpleFolder extends FolderMOAString {
  id_dp_operation: number;
  id_dossierprime: number;
  code_operation: string;
  moa?: MOA[];
  moe?: {}[];
  travaux?: {}[];
  moa_est_societe: 0;
  moa_is_syndic: BooleanNumber;
  moa_no_siret: BooleanNumber;
  moa_categorie_menage: number;
  moa_nombre_menage: number;
  moa_nombre_personne: number;
  moa_remp_preca: number;
  moe_denomination: string;
  moe_siret: string;
  moe_tel: string;
  moe_fax: string;
  moe_rue: string;
  moe_cp: string;
  moe_ville: string;
  moe_role: number;
  moe_commentaire: string;
  moe_individu_email: string;
  moe_contact: string;
  moe_contact_prenom: string;
  moe_contact_mobile: string;
  moe_contact_fonction: string;
  moe_bic: string;
  moe_iban: string;
  adresse_travaux_rue: string;
  adresse_travaux_nomsite: string;
  adresse_travaux_rue2: string;
  adresse_travaux_cp: string;
  adresse_travaux_ville: string;
  date_debut_travaux: string | null;
  date_fin_travaux: string | null;
  /* code_operation: "BAR-TH-106"
  date_reception: "2018-06-11"
  delai_instruction?: "331 jours de retard"
  id_dossierprime: 19925
  id_dp_operation: 60839
  is_avant_projet: 0
  label_public: "Dossier Incomplet"
  moa_denomination: ""
  moa_nom: "JAUD"
  moa_prenom: "Jean-Marie"
  statut_operation: 13 */
}

export interface Folder extends SimpleFolder {
  documents?: number[];
}

export interface FolderFull extends SimpleFolder {
  documents?: FileFull[];
}

export interface Folders {
  [index: number]: File;
}

export interface CheckPointCategory {
  id_point_controle_categorie: number;
  code_categorie: string;
  nom_categorie: string | null;
}

export interface CheckPointCategories {
  [index: number]: CheckPointCategory;
}

export interface CheckPoint {
  id_point_controle: number;
  id_categorie: number;
  nom: string;
  id_penalite: BooleanNumber;
  automatique: BooleanNumber;
  is_controle_file: BooleanNumber;
  pivot: {
    id_dp_operation: number;
    id_point_controle: number;
    valide: BooleanNumber;
    id_dp_file?: number;
  };
  pointcontrolcategories: CheckPointCategory;
}

export interface CheckPoints {
  [index: number]: CheckPoint;
}

export interface OperationSimple {
  id_dp_operation: number;
  id_dossierprime: number;
  id_operation: number;
  code_operation: string;
  statut?: {
    code_statut?: number;
    label_public?: string;
    id_statut?: number;
    label_prive?: string;
  };
  moderemuneration: {
    id_remuneration: number;
    delai_instruction: string;
  };
}

export interface Operation extends OperationSimple {
  dossierprime?: number;
  dossierprimefile?: [number];
  point_controles: [number];
}

export interface OperationFull extends OperationSimple {
  dossierprime?: FolderFull;
  dossierprimefile?: [FileFull];
  point_controles: [CheckPoint];
}

export interface Operations {
  [index: number]: Operation;
}

export interface Entities {
  files: Files;
  folders: Folders;
  checkPoints: CheckPoints;
  checkPointCategories: CheckPointCategories;
  operations: Operations;
  users: Users;
  comments: Comments;
}

export interface Normalized {
  entities: Entities;
}

export interface FilesFolderLoadedAction {
  type: typeof FOLDER_LOADED;
  normalized: Normalized;
}

export interface FilesFolcerCheckPointLoaded {
  type: typeof FOLDER_UPDATE_CHECK_POINT_LOADED;
  idDpFile: number;
  statusCode: number | null;
}

export interface FileLitigeLoaded {
  type: typeof FOLDER_FILE_LITIGE_LOADED;
  idDpFile: number;
  statusCode: number | null;
}

export interface FilesLogoutAction {
  type: typeof LOGOUT;
}

export type FilesActions = | FilesFolderLoadedAction
| FilesFolcerCheckPointLoaded
| FileLitigeLoaded
| FilesLogoutAction;

export interface FoldersFolderLoadedAction {
  type: typeof FOLDER_LOADED;
  normalized: Normalized;
}

export interface FoldersListLoadedAction {
  type: typeof LIST_LOADED;
  normalized: Normalized;
}

export interface FoldersLogoutAction {
  type: typeof LOGOUT;
}

export interface FoldersUpdateMoaLoaded {
  type: typeof FOLDER_UPDATE_MOA_LOADED;
  id_dossierprime: number;
  values: { [index: string]: string };
}

export type FoldersActions = | FoldersFolderLoadedAction
| FoldersLogoutAction
| FoldersUpdateMoaLoaded
| FoldersListLoadedAction;

export interface CheckPointsFolderLoadedAction {
  type: typeof FOLDER_LOADED;
  normalized: Normalized;
}

export interface CheckPointsListLoadedAction {
  type: typeof LIST_LOADED;
  normalized: Normalized;
}

export interface CheckPointsLogoutAction {
  type: typeof LOGOUT;
}

export interface CheckPointsFolderUpdateCheckpointLoadingAction {
  type: typeof FOLDER_UPDATE_CHECK_POINT_LOADING;
  checkPointId: number;
}

export interface CheckPointsFolderUpdateChekpointLoadedAction {
  type: typeof FOLDER_UPDATE_CHECK_POINT_LOADED;
  folderId: number;
  checkPointId: number;
}

export interface CheckPointsFolderUpdateCheckpointErrorAction {
  type: typeof FOLDER_UPDATE_CHECK_POINT_ERROR;
  checkPointId: number;
  preValue: BooleanNumber;
}

export type CheckPointsActions = | CheckPointsFolderUpdateCheckpointLoadingAction
| CheckPointsFolderUpdateChekpointLoadedAction
| CheckPointsFolderUpdateCheckpointErrorAction
| CheckPointsFolderLoadedAction
| CheckPointsListLoadedAction
| CheckPointsLogoutAction;

export interface CheckPointCategoriesFolderLoadedAction {
  type: typeof FOLDER_LOADED;
  normalized: Normalized;
}

export interface CheckPointCategoriesLogoutAction {
  type: typeof LOGOUT;
}

export type CheckPointCategoriesActions = | CheckPointCategoriesFolderLoadedAction
| CheckPointCategoriesLogoutAction;

export interface OperationsFolderLoadedAction {
  type: typeof FOLDER_LOADED;
  normalized: Normalized;
}
export interface OperationsListLoadedAction {
  type: typeof LIST_LOADED;
  normalized: Normalized;
}

export interface OperationsLogoutAction {
  type: typeof LOGOUT;
}

export type OperationsActions = | OperationsFolderLoadedAction
| OperationsListLoadedAction
| OperationsLogoutAction;

export interface UsersCommentsListLoadedAction {
  type: typeof COMMENTS_LIST_LOADED;
  normalized: Normalized;
}

export type UsersActions = UsersCommentsListLoadedAction;

export interface CommentsCommentsListLoadedAction {
  type: typeof COMMENTS_LIST_LOADED;
  normalized: Normalized;
}

export type CommentsActions = CommentsCommentsListLoadedAction;
