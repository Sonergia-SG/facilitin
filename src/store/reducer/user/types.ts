import {
  USER_ADD_TOKEN,
  LOGOUT,
  USER_INFOS_LOADING,
  USER_INFOS_LOADED,
  USER_INFOS_ERROR,
  INIT,
} from '../../types';

export type UserFonction = 'instructeur_initial' | 'instructeur_suivi' | '';

export interface UserInfos {
  nom: string;
  prenom: string;
  identifiant: string;
  id_groupe: number;
  fonction: UserFonction;
}

export interface UserState {
  apiKey: string | null;
  infosLoading: boolean;
  user: UserInfos | null;
}

export interface UserInit {
  type: typeof INIT;
}

export interface UserAddTokenAction {
  type: typeof USER_ADD_TOKEN;
  apiKey: string | null;
}

export interface UserInfosLoading {
  type: typeof USER_INFOS_LOADING;
}

export interface UserInfosLoaded {
  type: typeof USER_INFOS_LOADED;
  user: UserInfos;
}

export interface UserInfosError {
  type: typeof USER_INFOS_ERROR;
}

export interface UserLogoutAction {
  type: typeof LOGOUT;
}

export type UserActionTypes = | UserInit
  | UserAddTokenAction
  | UserLogoutAction
  | UserInfosLoading
  | UserInfosLoaded
  | UserInfosError;
