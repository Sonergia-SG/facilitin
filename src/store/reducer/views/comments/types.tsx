import {
  INIT,
  COMMENTS_LIST_LOADING,
  COMMENTS_LIST_LOADED,
  COMMENTS_LIST_ERROR,
  UPDATE_NEW_COMMENT_MESSAGE,
  COMMENTS_POST_COMMENT_LOADING,
  COMMENTS_POST_COMMENT_LOADED,
  COMMENTS_POST_COMMENT_ERROR,
} from '../../../types';
import { CommentsLoadedNormalized } from '../../../actions/views/comments';

export interface CommentsByFoldersPending {
  newMessage: string;
  loading: boolean;
  postLoading: boolean;
}

export interface CommentsByFolders {
  pending: CommentsByFoldersPending;
  comments: Array<number>;
}

export interface CommentsState {
  byFolders: {
    [index: number]: CommentsByFolders;
  };
}

interface CommentsIniAction {
  type: typeof INIT;
}

export interface CommentsListLoadingAction {
  type: typeof COMMENTS_LIST_LOADING;
  idDpFolder: number;
}

export interface CommentsListLoadedAction {
  type: typeof COMMENTS_LIST_LOADED;
  idDpFolder: number;
  normalized: CommentsLoadedNormalized;
}

export interface CommentsListErrorAction {
  type: typeof COMMENTS_LIST_ERROR;
  idDpFolder: number;
}

export interface CommentsUpdateNewMessage {
  type: typeof UPDATE_NEW_COMMENT_MESSAGE;
  message: string;
  idDpFolder: number;
}

export interface CommentsPostCommentLoading {
  type: typeof COMMENTS_POST_COMMENT_LOADING;
  idDpFolder: number;
}

export interface CommentsPostCommentLoaded {
  type: typeof COMMENTS_POST_COMMENT_LOADED;
  idDpFolder: number;
}

export interface CommentsPostCommentError {
  type: typeof COMMENTS_POST_COMMENT_ERROR;
  idDpFolder: number;
}

export type CommentsAction = | CommentsIniAction
| CommentsListLoadingAction
| CommentsListLoadedAction
| CommentsListErrorAction
| CommentsUpdateNewMessage
| CommentsPostCommentLoading
| CommentsPostCommentLoaded
| CommentsPostCommentError;
