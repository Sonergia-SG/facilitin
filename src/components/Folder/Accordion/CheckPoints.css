.CheckPoints-CheckPoint-Container {
  margin-left: 2px;
}

.CheckPoints-Table th {
  padding: 0 5px;
  font-size: 12px;
}

.CheckPoints-Table .CheckPoints-Table-Center {
  text-align: center;
}

.CheckPoints-CheckPoint-Label {
  margin-left: 5px;
  font-size: 11px;
  line-height: 0%;
}
