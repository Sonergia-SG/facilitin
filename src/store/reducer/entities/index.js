import { LIST_LOADED } from '../../types';

const initialState = {
  folders: {},
  documents: {},
  moa: {},
  moe: {},
  travaux: {},
};

const entities = (state = initialState, action) => {
  switch (action.type) {
    case LIST_LOADED:
      return {
        ...state,
        folders: {
          ...state.folders,
          ...action.normalized.entities.folders,
        },
      };
    default:
      return state;
  }
};

export default entities;
