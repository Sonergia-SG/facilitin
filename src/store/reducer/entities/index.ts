import { combineReducers } from 'redux';

import folders from './folders';
import checkPoints from './checkPoints';
import files from './files';
import moa from './moa';
import moe from './moe';
import travaux from './travaux';

export default combineReducers({
  folders,
  moa,
  moe,
  travaux,
  checkPoints,
  files,
});
