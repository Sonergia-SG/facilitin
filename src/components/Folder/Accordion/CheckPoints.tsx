import React from 'react';
import { connect } from 'react-redux';
import idx from 'idx';

import { CheckPoint } from '../../../store/reducer/entities/types';
import { updateFolderCheckPoint } from '../../../store/actions/views/folder';

import './CheckPoints.css';
import { FolderPendingItem } from '../../../store/reducer/views/folder/types';

import Picto from './Picto';

interface Props {
  checkPoints: Array<CheckPoint> | undefined;
  fileId: number;
  folderId: number;
  updateCheckPoint: any;
  pending: FolderPendingItem | undefined;
}

export const CheckPointsComponent = ({
  checkPoints,
  fileId,
  folderId,
  updateCheckPoint,
  pending,
}: Props) => {
  const fileCheckPoints = (checkPoints || []).filter(c => c.pivot.id_dp_file === fileId);

  if (fileCheckPoints.length === 0) {
    return (
      <div className="CheckPoints-CheckPoint-Container">
        <h4>Pas de points de controles pour ce fichier</h4>
      </div>
    );
  }

  const loading = pending ? !!pending.loading : true;

  return (
    <div className="CheckPoints-CheckPoint-Container">
      <table className="CheckPoints-Table">
        <thead>
          <tr>
            <th>Oui</th>
            <th>Non</th>
            <th />
            <th />
          </tr>
        </thead>
        <tbody>
          {fileCheckPoints.map((value) => {
            const checkPointStatus = idx(
              pending,
              _ => _.checkPoint[value.id_point_controle].status,
            );

            const disabled = loading || checkPointStatus === 'SENDING' || value.automatique === 1;

            return (
              <tr key={value.id_point_controle}>
                <td className="CheckPoints-Table-Center">
                  <input
                    type="radio"
                    id={`${value.is_controle_file}_yes`}
                    name={`${value.id_point_controle}`}
                    checked={value.pivot.valide === 1}
                    value="yes"
                    disabled={disabled}
                    onChange={() => {
                      updateCheckPoint({
                        folderId,
                        checkPointId: value.id_point_controle,
                        idDpFile: value.pivot.id_dp_file,
                        newValue: 1,
                      });
                    }}
                  />
                </td>
                <td className="CheckPoints-Table-Center">
                  <input
                    type="radio"
                    id={`${value.is_controle_file}_no`}
                    name={`${value.id_point_controle}`}
                    checked={value.pivot.valide === 0}
                    value="no"
                    disabled={disabled}
                    onChange={() => {
                      updateCheckPoint({
                        folderId,
                        checkPointId: value.id_point_controle,
                        idDpFile: value.pivot.id_dp_file,
                        newValue: 0,
                      });
                    }}
                  />
                </td>
                <td className="CheckPoints-Table-Center">
                  <Picto checkPoint={value} />
                </td>
                <td>
                  <label className="CheckPoints-CheckPoint-Label" htmlFor={'{value.id_controle}'}>
                    {value.nom}
                  </label>
                </td>
              </tr>
            );
          })}
        </tbody>
      </table>
    </div>
  );
};

export default connect(
  null,
  { updateCheckPoint: updateFolderCheckPoint },
)(CheckPointsComponent);
