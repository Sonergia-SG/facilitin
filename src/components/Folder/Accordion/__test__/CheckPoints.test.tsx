import React from 'react';
import { shallow } from 'enzyme';

import { CheckPointsComponent as CheckPoints } from '../CheckPoints';
import { CheckPoint } from '../../../../store/reducer/entities/types';

const defaultCheckPoint: CheckPoint = {
  id_point_controle: 0,
  id_categorie: 0,
  nom: 'Checkpoint test',
  id_penalite: 1,
  automatique: 0,
  is_controle_file: 0,
  pivot: {
    id_dp_operation: 0,
    id_point_controle: 0,
    valide: 0,
    id_dp_file: 0,
  },
  pointcontrolcategories: {
    id_point_controle_categorie: 0,
    code_categorie: 'CAT',
    nom_categorie: 'NAME',
  },
};

describe('CheckPoints', () => {
  it('render correctly with default checkpoint', () => {
    const wrapper = shallow(
      <CheckPoints
        checkPoints={[defaultCheckPoint]}
        fileId={0}
        folderId={0}
        updateCheckPoint={() => 'Hello'}
        locked={false}
        pending={{}}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
  it('render correctly when locked', () => {
    const wrapper = shallow(
      <CheckPoints
        checkPoints={[defaultCheckPoint]}
        fileId={0}
        folderId={0}
        updateCheckPoint={() => 'Hello'}
        locked
        pending={{}}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
  it('render correctly with checkpoint valid', () => {
    const wrapper = shallow(
      <CheckPoints
        checkPoints={[{
          ...defaultCheckPoint,
          pivot: {
            ...defaultCheckPoint.pivot,
            valide: 1,
          },
        }]}
        fileId={0}
        folderId={0}
        updateCheckPoint={() => 'Hello'}
        locked={false}
        pending={{}}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
  it('render correctly with checkpoint no traited', () => {
    const wrapper = shallow(
      <CheckPoints
        checkPoints={[{
          ...defaultCheckPoint,
          pivot: {
            ...defaultCheckPoint.pivot,
            valide: -1,
          },
        }]}
        fileId={0}
        folderId={0}
        updateCheckPoint={() => 'Hello'}
        locked={false}
        pending={{}}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('render correctly with checkpoint sending', () => {
    const wrapper = shallow(
      <CheckPoints
        checkPoints={[{
          ...defaultCheckPoint,
          pivot: {
            ...defaultCheckPoint.pivot,
            valide: 1,
          },
        }]}
        fileId={0}
        folderId={0}
        updateCheckPoint={() => 'Hello'}
        locked={false}
        pending={{
          checkPoint: {
            0: {
              status: 'SENDING',
            },
          },
        }}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('render correctly during folder loading', () => {
    const wrapper = shallow(
      <CheckPoints
        checkPoints={[{
          ...defaultCheckPoint,
          pivot: {
            ...defaultCheckPoint.pivot,
            valide: 1,
          },
        }]}
        fileId={0}
        folderId={0}
        updateCheckPoint={() => 'Hello'}
        locked={false}
        pending={{
          loading: true,
        }}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('can\'t update checkpoint if it\'s an automatic checkpoint', () => {
    const wrapper = shallow(
      <CheckPoints
        checkPoints={[{
          ...defaultCheckPoint,
          automatique: 1,
        }]}
        fileId={0}
        folderId={0}
        updateCheckPoint={() => 'Hello'}
        locked={false}
        pending={{}}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
