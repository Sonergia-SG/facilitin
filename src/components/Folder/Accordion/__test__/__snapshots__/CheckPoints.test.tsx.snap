// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckPoints can't update checkpoint if it's an automatic checkpoint 1`] = `
<div
  className="CheckPoints-CheckPoint-Container"
>
  <table
    className="CheckPoints-Table"
  >
    <thead>
      <tr>
        <th>
          Oui
        </th>
        <th>
          Non
        </th>
        <th />
      </tr>
    </thead>
    <tbody>
      <tr
        key="0"
      >
        <td
          className="CheckPoints-Table-Center"
        >
          <Radio
            checked={false}
            disabled={true}
            id="0_yes"
            name="0"
            onChange={[Function]}
            value="yes"
          />
        </td>
        <td
          className="CheckPoints-Table-Center"
        >
          <Radio
            checked={true}
            customColor="#FF6C60"
            disabled={true}
            id="0_no"
            name="0"
            onChange={[Function]}
            value="no"
          />
        </td>
        <td>
          <label
            className="CheckPoints-CheckPoint-Label"
            htmlFor="{value.id_controle}"
          >
            Checkpoint test
          </label>
        </td>
        <Modal
          actions={
            Object {
              "cancel": Object {
                "handle": [Function],
                "title": "Annuler",
              },
              "confirm": Object {
                "handle": [Function],
                "title": "Rejet du document",
              },
              "type": "confirm",
            }
          }
          displayModal={false}
          message="La non-validation de ce point de contrôle entraine un rejet du document undefined, êtes-vous certain de vouloir continuer ?"
          title="Point de contrôle non conforme"
        />
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`CheckPoints render correctly during folder loading 1`] = `
<div
  className="CheckPoints-CheckPoint-Container"
>
  <table
    className="CheckPoints-Table"
  >
    <thead>
      <tr>
        <th>
          Oui
        </th>
        <th>
          Non
        </th>
        <th />
      </tr>
    </thead>
    <tbody>
      <tr
        key="0"
      >
        <td
          className="CheckPoints-Table-Center"
        >
          <Radio
            checked={true}
            disabled={true}
            id="0_yes"
            name="0"
            onChange={[Function]}
            value="yes"
          />
        </td>
        <td
          className="CheckPoints-Table-Center"
        >
          <Radio
            checked={false}
            customColor="#FCB322"
            disabled={true}
            id="0_no"
            name="0"
            onChange={[Function]}
            value="no"
          />
        </td>
        <td>
          <label
            className="CheckPoints-CheckPoint-Label"
            htmlFor="{value.id_controle}"
          >
            Checkpoint test
          </label>
        </td>
        <Modal
          actions={
            Object {
              "cancel": Object {
                "handle": [Function],
                "title": "Annuler",
              },
              "confirm": Object {
                "handle": [Function],
                "title": "Rejet du document",
              },
              "type": "confirm",
            }
          }
          displayModal={false}
          message="La non-validation de ce point de contrôle entraine un rejet du document undefined, êtes-vous certain de vouloir continuer ?"
          title="Point de contrôle non conforme"
        />
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`CheckPoints render correctly when locked 1`] = `
<div
  className="CheckPoints-CheckPoint-Container"
>
  <table
    className="CheckPoints-Table"
  >
    <thead>
      <tr>
        <th>
          Oui
        </th>
        <th>
          Non
        </th>
        <th />
      </tr>
    </thead>
    <tbody>
      <tr
        key="0"
      >
        <td
          className="CheckPoints-Table-Center"
        >
          <Radio
            checked={false}
            disabled={true}
            id="0_yes"
            name="0"
            onChange={[Function]}
            value="yes"
          />
        </td>
        <td
          className="CheckPoints-Table-Center"
        >
          <Radio
            checked={true}
            customColor="#FF6C60"
            disabled={true}
            id="0_no"
            name="0"
            onChange={[Function]}
            value="no"
          />
        </td>
        <td>
          <label
            className="CheckPoints-CheckPoint-Label"
            htmlFor="{value.id_controle}"
          >
            Checkpoint test
          </label>
        </td>
        <Modal
          actions={
            Object {
              "cancel": Object {
                "handle": [Function],
                "title": "Annuler",
              },
              "confirm": Object {
                "handle": [Function],
                "title": "Rejet du document",
              },
              "type": "confirm",
            }
          }
          displayModal={false}
          message="La non-validation de ce point de contrôle entraine un rejet du document undefined, êtes-vous certain de vouloir continuer ?"
          title="Point de contrôle non conforme"
        />
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`CheckPoints render correctly with checkpoint no traited 1`] = `
<div
  className="CheckPoints-CheckPoint-Container"
>
  <table
    className="CheckPoints-Table"
  >
    <thead>
      <tr>
        <th>
          Oui
        </th>
        <th>
          Non
        </th>
        <th />
      </tr>
    </thead>
    <tbody>
      <tr
        key="0"
      >
        <td
          className="CheckPoints-Table-Center"
        >
          <Radio
            checked={false}
            disabled={false}
            id="0_yes"
            name="0"
            onChange={[Function]}
            value="yes"
          />
        </td>
        <td
          className="CheckPoints-Table-Center"
        >
          <Radio
            checked={false}
            customColor="#FCB322"
            disabled={false}
            id="0_no"
            name="0"
            onChange={[Function]}
            value="no"
          />
        </td>
        <td>
          <label
            className="CheckPoints-CheckPoint-Label"
            htmlFor="{value.id_controle}"
          >
            Checkpoint test
          </label>
        </td>
        <Modal
          actions={
            Object {
              "cancel": Object {
                "handle": [Function],
                "title": "Annuler",
              },
              "confirm": Object {
                "handle": [Function],
                "title": "Rejet du document",
              },
              "type": "confirm",
            }
          }
          displayModal={false}
          message="La non-validation de ce point de contrôle entraine un rejet du document undefined, êtes-vous certain de vouloir continuer ?"
          title="Point de contrôle non conforme"
        />
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`CheckPoints render correctly with checkpoint sending 1`] = `
<div
  className="CheckPoints-CheckPoint-Container"
>
  <table
    className="CheckPoints-Table"
  >
    <thead>
      <tr>
        <th>
          Oui
        </th>
        <th>
          Non
        </th>
        <th />
      </tr>
    </thead>
    <tbody>
      <tr
        key="0"
      >
        <td
          className="CheckPoints-Table-Center"
        >
          <Radio
            checked={true}
            disabled={true}
            id="0_yes"
            name="0"
            onChange={[Function]}
            value="yes"
          />
        </td>
        <td
          className="CheckPoints-Table-Center"
        >
          <Radio
            checked={false}
            customColor="#FCB322"
            disabled={true}
            id="0_no"
            name="0"
            onChange={[Function]}
            value="no"
          />
        </td>
        <td>
          <label
            className="CheckPoints-CheckPoint-Label"
            htmlFor="{value.id_controle}"
          >
            Checkpoint test
          </label>
        </td>
        <Modal
          actions={
            Object {
              "cancel": Object {
                "handle": [Function],
                "title": "Annuler",
              },
              "confirm": Object {
                "handle": [Function],
                "title": "Rejet du document",
              },
              "type": "confirm",
            }
          }
          displayModal={false}
          message="La non-validation de ce point de contrôle entraine un rejet du document undefined, êtes-vous certain de vouloir continuer ?"
          title="Point de contrôle non conforme"
        />
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`CheckPoints render correctly with checkpoint valid 1`] = `
<div
  className="CheckPoints-CheckPoint-Container"
>
  <table
    className="CheckPoints-Table"
  >
    <thead>
      <tr>
        <th>
          Oui
        </th>
        <th>
          Non
        </th>
        <th />
      </tr>
    </thead>
    <tbody>
      <tr
        key="0"
      >
        <td
          className="CheckPoints-Table-Center"
        >
          <Radio
            checked={true}
            disabled={false}
            id="0_yes"
            name="0"
            onChange={[Function]}
            value="yes"
          />
        </td>
        <td
          className="CheckPoints-Table-Center"
        >
          <Radio
            checked={false}
            customColor="#FCB322"
            disabled={false}
            id="0_no"
            name="0"
            onChange={[Function]}
            value="no"
          />
        </td>
        <td>
          <label
            className="CheckPoints-CheckPoint-Label"
            htmlFor="{value.id_controle}"
          >
            Checkpoint test
          </label>
        </td>
        <Modal
          actions={
            Object {
              "cancel": Object {
                "handle": [Function],
                "title": "Annuler",
              },
              "confirm": Object {
                "handle": [Function],
                "title": "Rejet du document",
              },
              "type": "confirm",
            }
          }
          displayModal={false}
          message="La non-validation de ce point de contrôle entraine un rejet du document undefined, êtes-vous certain de vouloir continuer ?"
          title="Point de contrôle non conforme"
        />
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`CheckPoints render correctly with default checkpoint 1`] = `
<div
  className="CheckPoints-CheckPoint-Container"
>
  <table
    className="CheckPoints-Table"
  >
    <thead>
      <tr>
        <th>
          Oui
        </th>
        <th>
          Non
        </th>
        <th />
      </tr>
    </thead>
    <tbody>
      <tr
        key="0"
      >
        <td
          className="CheckPoints-Table-Center"
        >
          <Radio
            checked={false}
            disabled={false}
            id="0_yes"
            name="0"
            onChange={[Function]}
            value="yes"
          />
        </td>
        <td
          className="CheckPoints-Table-Center"
        >
          <Radio
            checked={true}
            customColor="#FF6C60"
            disabled={false}
            id="0_no"
            name="0"
            onChange={[Function]}
            value="no"
          />
        </td>
        <td>
          <label
            className="CheckPoints-CheckPoint-Label"
            htmlFor="{value.id_controle}"
          >
            Checkpoint test
          </label>
        </td>
        <Modal
          actions={
            Object {
              "cancel": Object {
                "handle": [Function],
                "title": "Annuler",
              },
              "confirm": Object {
                "handle": [Function],
                "title": "Rejet du document",
              },
              "type": "confirm",
            }
          }
          displayModal={false}
          message="La non-validation de ce point de contrôle entraine un rejet du document undefined, êtes-vous certain de vouloir continuer ?"
          title="Point de contrôle non conforme"
        />
      </tr>
    </tbody>
  </table>
</div>
`;
